config.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import random
+import json
+import yaml
+import sys
+import os
+
+from dotmap import DotMap
+
+# This is the minimal development config needed.
+DEFAULT_DEV_CONFIG = {
+    "twitch": {
+        "enabled": False,
+        "oauth_token": "<Get one at https://twitchapps.com/tmi/>",
+        "channel": "baalorlord",
+    },
+    "discord": {
+        "enabled": False,
+        "oauth_token": "<token unset>"
+    },
+    "server": {"debug": False, "secret": "<i-haven't-set-a-secret>"},
+}
+
+def load_file(filename: str) -> DotMap:
+    with open(filename, 'r') as stream:
+        # With _dynamic=False, any access on something we hadn't yet
+        # set will raise an exception. With _dynamic=True (the default) it
+        # would instead return None. This lets us know when we access things
+        # that aren't set.
+        conf = DotMap(yaml.safe_load(stream), _dynamic=False)
+        return conf
+
+# Load the default configuration file
+config = load_file("./default-config.yml")
```

Comment:
Python noob question: does this have types and properties? 

If not: could we have that?

Reply:
It follows whatever types we've set in the configuration file, but there's no validation or anything like that. I personally don't think that's super important for the configuration files, especially as long as the default file has the right things.

I also saw that you made #27 only after I had made this PR, haha.